mjidris/irondb
# Us 241

## Objective
Clean up `iron.sh` installation to work more reliably. This will set us up for connecting the React front-end to the API, and ultimately interact with the data like we need.

## Detailed Description
Most notably, initialization scripts in the Postgres container weren't working as expected. Work was done to adjust `iron.sh`, `docker-compose.yml` and the SQL scripts themselves so that everything would be built properly upon installation.

## Screenshot (if applicable)


## How to Test (if applicable)
Pull down this branch and then run through the following instructions.
First, make sure all of you Docker environment is totally clean:
1. Start with `./iron.sh -s` in the project root. 
2. Run `docker volume prune`.
3. Run `docker image prune`.
4. Run `docker image rm irondb_pgdb`

After your environment is cleaned up, run `./iron.sh -i`. After everything is rebuilt, run `docker ps`, and all of the containers should be running. Navigate to `localhost:3001`. The styles will most-likely be totally broken, but if you don't get "Internal server error", that means everything is working.

## Tests (if added)


## Requirements
[US-241](https://tree.taiga.io/project/cbrobsto-iron-db/us/241)
[T-242](https://tree.taiga.io/project/cbrobsto-iron-db/task/242)
[T-248](https://tree.taiga.io/project/cbrobsto-iron-db/task/248)
